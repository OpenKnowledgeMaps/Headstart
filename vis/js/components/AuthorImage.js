import React from "react";
import { connect } from "react-redux";

import AuthorImageTemplate from "../templates/AuthorImage";

import defaultImage from "../../images/author_default.png";

export const AuthorImage = ({ hidden, imageLink = "" }) => {
  if (hidden) {
    return null;
  }

  let link = defaultImage;
  if (imageLink !== "") {
    link = imageLink;
  }

  return <AuthorImageTemplate link={link} />;
};

const mapStateToProps = (state) => ({
  hidden: !state.contextLine.showAuthor,
  imageLink: state.contextLine.author.imageLink,
});

export default connect(mapStateToProps)(AuthorImage);
