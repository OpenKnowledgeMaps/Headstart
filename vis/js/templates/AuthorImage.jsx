import React from "react";

import defaultImage from "../../images/author_default.png";

const AuthorImage = ({ url = "" }) => {
  let link = defaultImage;
  if (url !== "") {
    link = url;
  }

  return (
    // html template starts here
    <div id="title_image" className="titleimage">
      <a id="author_image_link" href={link} target="_blank">
        <div
          id="author_image"
          className="authorimage"
          style={{ backgroundImage: `url(${link})` }}
        ></div>
      </a>
    </div>
    // html template ends here
  );
};

export default AuthorImage;
