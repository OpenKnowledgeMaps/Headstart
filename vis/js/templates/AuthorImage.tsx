import React from "react";

// @ts-ignore
import defaultImage from "../../images/author_default.png";
import { ServiceType } from "../@types/service";

export interface AuthorImageProps {
  service: ServiceType;
  url: string;
  orcidId: string;
}

const AuthorImage = ({ service, url = "", orcidId }: AuthorImageProps) => {
  let link = defaultImage;
  
  if (url) {
    link = url;
  }

  let href = "";

  if (service === ServiceType.ORCID) {
    href = `https://orcid.org/${orcidId}`
  } else {
    href = link;
  }

  return (
    <div id="title_image" className="titleimage">
      <a
        id="author_image_link"
        href={href}
        target="_blank"
        rel="noreferrer"
        aria-label="author image"
        // if href is empty, then the link is not clickable
        style={{ pointerEvents: href ? "auto" : "none" }}
      >
        <div
          id="author_image"
          className="authorimage"
          style={{ backgroundImage: `url(${link})` }}
          role="img"
          aria-label="author image"
        ></div>
      </a>
    </div>
  );
};

export default AuthorImage;
