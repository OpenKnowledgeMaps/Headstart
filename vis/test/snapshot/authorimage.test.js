import React from "react";
import renderer from "react-test-renderer";

import AuthorImage from "../../js/templates/AuthorImage";

describe("Author image component snapshot", () => {
  it("matches as author image snapshot", () => {
    const tree = renderer
      .create(<AuthorImage url="https://example.com/12345" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
