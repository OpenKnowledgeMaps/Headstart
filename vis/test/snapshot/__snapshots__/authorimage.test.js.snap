// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author image component snapshot matches as author image snapshot 1`] = `
<div
  className="titleimage"
  id="title_image"
>
  <a
    href="https://example.com/12345"
    id="author_image_link"
    target="_blank"
  >
    <div
      className="authorimage"
      id="author_image"
      style={
        Object {
          "backgroundImage": "url(https://example.com/12345)",
        }
      }
    />
  </a>
</div>
`;
